tests/acceptance/test_med_safety_dsl.py

```diff
@@ -0,0 +1,289 @@
+
+    with subtest(
+        "practice_id", missing_feature="add practice ids to practice registrations"
+    ):
+        cohort.practice_id = (
+            filtered_practice_registrations.sort_by(practice_registrations.date_end)
+            .last_for_patient()
+            .select_column(practice_registrations.practice_id)
+        )
+        cohort.assert_results(practice_id=1)
+
+    with subtest("age"):
+        cohort.age = (
+            index_date
+            - patient_demographics.select_column(patient_demographics.date_of_birth)
+        ).convert_to_years()
+        cohort.assert_results(age=65)
+
+    with subtest("sex"):
+        cohort.sex = patient_demographics.select_column(patient_demographics.sex)
+        cohort.assert_results(sex="female")
+
+    # Indicators
+    # ==========
+
+    # GIB01
+    # -----
+    with subtest("oral_nsaid"):
+        cohort.oral_nsaid = (
+            clinical_events.filter(clinical_events.code.is_in(oral_nsaid_codelist))
+            .filter(clinical_events.date >= index_date_minus_3_months)
+            .filter(clinical_events.date <= index_date)
+            .exists_for_patient()
+        )
+        cohort.assert_results(oral_nsaid=True)
+
+    with subtest("ppi"):
+        cohort.ppi = (
+            clinical_events.filter(clinical_events.code.is_in(ppi_codelist))
+            .filter(clinical_events.date >= index_date_minus_3_months)
+            .filter(clinical_events.date <= index_date)
+            .exists_for_patient()
+        )
+        cohort.assert_results(ppi=True)
+
+    with subtest("indicator_gib_risk", missing_feature="Arithmetic on BoolSeries"):
+        at_gib_risk = cohort.oral_nsaid & (cohort.age >= 65)  # noqa: F841
+        not_at_gib_risk = at_gib_risk & ~cohort.ppi  # noqa: F841
+
+        # Why does the following numerator/denominator pair need to be commented out (to
+        # avoid raising a TypeError) but the other two numerator/denominator pairs do
+        # not?
+        # cohort.indicator_gib_risk_numerator = not_at_gib_risk
+        # cohort.indicator_gib_risk_denominator = at_gib_risk
+
+        cohort.assert_results(indicator_gib_risk_numerator=None)
+        cohort.assert_results(indicator_gib_risk_denominator=None)
+
+    with subtest(
+        "indicator_gib_num_admissions",
+        missing_feature="Missing TPPBackend.hospital_admissions",
+    ):
+        # TPPBackend.hospitalizations exists but doesn't expose an admission_method
+        # property. It's also not clear whether hospitalizations.code ==
+        # hospital_admissions.primary_diagnosis.
+        cohort.indicator_gib_num_admissions = (
+            hospital_admissions.filter(
+                hospital_admissions.primary_diagnosis.is_in(gib_admissions_codelist)
+            )
+            .filter(
+                # TODO: Query an IntColumn with is_in
+                hospital_admissions.admission_method.is_in(
+                    EMERGENCY_ADMISSIONS_CODELIST
+                )
```

Comment:
Can you explain what this "codelist" is? If the column is an int, I guess it's not a normal codelist.

Reply:
Thanks for picking me up on this, as I had copied it from the v1 study definition uncritically. There's more information in [this Slack thread][1] about why we don't represent admission method codes as a codelist. See 5819ac8.

[1]: https://ebmdatalab.slack.com/archives/C31D62X5X/p1643797768199749